File: PROGRAMMERS/LEVEL1/3진법 뒤집기.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int n) {
    int answer = 0;
    vector<int> v;
    while (n) {
        v.push_back(n % 3);
        n /= 3;
    }
    int j = 1;
    int m = v.size();
    for (int i = 0; i < m; i++) {
        answer += v.back() * j;
        v.pop_back();
        j *= 3;
    }
    return answer;
}

File: PROGRAMMERS/LEVEL1/3진법 뒤집기.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.*;

class Solution {
    public int solution(int n) {
        int answer = 0;
        
        List<Integer> tmp = new ArrayList<Integer>();
        
        while(n>0) {
        	tmp.add(n%3);
        	n/=3;
        }
        
        int mul=1;
        for(int i=tmp.size()-1;i>=0;i--) {
        	answer+=mul*tmp.get(i);
        	mul*=3;
        }
        
        return answer;
    }
}